Makefile

```diff
@@ -1,74 +1,88 @@
 #
-# Work in progress makefile to simplify Figures development
+# Makefile to simplify Figures development
 #
 
-.PHONY: help clean_tests clean_webpack_buld clean_python_build build_python lint pip_install test twine_check twine_push_test delete_mock_migrations reset_mock_migrations
-
-help:
-	@echo "Targets:"
-	@echo " - clean_python"
-	@echo " - clean_python_build"
-	@echo " - clean_webpack_build"
-	@echo " - build_python"
-	@echo " - coverage"
-	@echo " - lint"
-	@echo " - pip_install"
-	@echo " - test"
-	@echo " - twine_check"
-	@echo " - twine_push_test"
-	@echo " - twine_push_prod"
-	@echo " - delete_mock_migrations"
-	@echo " - update_mock_migrations"
-
-clean_python:
+include .env*
+export $(shell sed 's/=.*//' .env*)
+
+SHELL := /bin/bash
+.PHONY: help
+
+
+help: ## This help message
+	@echo -e "$$(grep -hE '^\S+:.*##' $(MAKEFILE_LIST) | \
+		sed -e 's/:.*##\s*/:/' -e 's/^\(.\+\):\(.*\)/\\x1b[36m\1\\x1b[m:\2/' | column -c2 -t -s :)"
+
+environment.display:  ## Prints the environment values used in this makefile as defined in .env.* files
+	@echo EDX_PLATFORM_RELEASE = $(EDX_PLATFORM_RELEASE)
+
+
+##
+## Python targets
+##
+
+python.clean:  ## Removes generated Python bytecode files
+	@echo Removing generated Python bytecode files...
 	find tests -type f -name "*.pyc" -exec rm -f {} \;
 	find figures -type f -name "*.pyc" -exec rm -f {} \;
 	find devsite -type f -name "*.pyc" -exec rm -f {} \;
+	find mocks -type f -name "*.pyc" -exec rm -f {} \;
 	find tests -type d -name __pycache__ -exec rm -r {} \+
 	find devsite -type d -name __pycache__ -exec rm -r {} \+
+	find mocks -type d -name __pycache__ -exec rm -r {} \+
 
 # Clean the Python dist build
-clean_python_build:
+python.build.clean:  ## Removes Python packaging build files
 	rm -rf dist
 	rm -rf Figures.egg-info
 
-# Clean the webpack build
-clean_webpack_build:
-	rm -rf figures/static/figures/*
-	rm figures/webpack-stats.json
+pip.install:   ## Install Appsembler Hawthorn requirements for devsite
+	pip install -r devsite/requirements/hawthorn_appsembler.txt
+
 
-build_python: clean_python_build
+python.build:   python.build.clean  ## Build Python package
 	python setup.py sdist bdist_wheel --universal
 
-coverage:
-	py.test --cov-report term-missing --cov=figures tests/
 
-lint:
+pylint:  ## lint
 	pylint --load-plugins pylint_django ./figures
 
-pip_install:
-	pip install -r devsite/requirements/hawthorn.txt
+coverage:  ## Run coverage, without the built-in virtualenv
+	@. ve/bin/activate; coverage run --source figures -m py.test; coverage report -m
+
+
+### Automatically constructed Virtualenv based targets
+
+ve/bin/figures-ws: devsite/requirements/${EDX_PLATFORM_RELEASE}_appsembler.txt
```

Comment:
Make magic! I didn't know it can work on paths. Thanks!

Reply:
Cool, huh @OmarIthawi? I learned that one from @ahmedaljazzar